### ISSUE TITLE
FAILED akka.stream.scaladsl.HubSpec

### ISSUE THREAD

**Issue by johanandren** [akka/akka#21391]
https://jenkins.akka.io:8498/job/akka-nightly/3666/consoleFull

```
[info] - must send the same elements to consumers attaching around the same time *** FAILED *** (7 seconds, 151 milliseconds)
[info]   A timeout occurred waiting for a future to complete. Queried 333 times, sleeping 15 milliseconds between each query. (HubSpec.scala:199)
[info]   org.scalatest.concurrent.Futures$FutureConcept$$anon$1:
[info]   at org.scalatest.concurrent.Futures$FutureConcept$class.tryTryAgain$1(Futures.scala:538)
[info]   at org.scalatest.concurrent.Futures$FutureConcept$class.futureValueImpl(Futures.scala:550)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
[info]   at org.scalatest.concurrent.Futures$FutureConcept$class.futureValue(Futures.scala:476)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValue(ScalaFutures.scala:275)
[info]   at akka.stream.scaladsl.HubSpec$$anonfun$2$$anonfun$apply$mcV$sp$17$$anonfun$apply$15.apply(HubSpec.scala:199)
```

**Comment by drewhk** [comment]
Yaay, this looks tricky :(

**Comment by drewhk** [comment]
Hm, thankfully there are Graphviz Outputs!!!! It seems like the BroadcastHub "shouldPull" the upstream sequence while at the same time there is a downstream that it "shouldPush". I.e. this looks like a deadlock of some sort.

**Comment by drewhk** [comment]
would be nice to think that https://github.com/akka/akka/pull/21447 is related, but I don't see how it could be related. Unless there was an unreported exception that caused cancellation or failure somewhere, in which case it can be. But Detacher is still a mistery for me in that diagram...

**Comment by drewhk** [comment]
Eh, Detacher can be in that state if it already buffered up an element. OK, so then the question is why the Hub stalled. Seems like one downstream is finished already, the other is waiting an element.

**Comment by drewhk** [comment]
Ok, one more clue. There is only one downstream consumer reported. This is weird as both of them have been materialized and the stream could not have been completed (StatefulMapConcat is still alive). So I suspect that one of the consumers might have failed and left the Hub in a backpressured state. This means that the first assertion fails on a timeout *before* the second assertion could report the initialization error (hence we don't see it)

**Comment by drewhk** [comment]
The build failure was reported on Sep 9 by Jenkins and this was fixed on 13th: https://github.com/akka/akka/pull/21443

That bug could explain the situation here (per my previous comment) and would also explain why this have not failed before. Since that bug failed on hubCallback being `null`, that would explain why it was not unregistered in `postStop()`: that also needs a non-null hubCallback.

TLDR: I consider this fixed :D
